### DISCUSSION TITLE
Bad Gateway. 502 Error Code

### DISCUSSION THREAD

**Discussion by maciejbuchert** [homeassistant-apps/app-cloudflared#374] [Q&A] [answered]
### The problem

I've tried local IP, hostname (homeassistant and hassio) as service IP.
I've added `172.30.33.0/24` to trusted_proxies
I've configured external url for home assistant

Here is CF config
![Zrzut ekranu 2023-04-13 o 13 40 40](https://user-images.githubusercontent.com/9607667/231748487-582369a2-7949-4e23-8cd7-f9def91788db.png)

### What version of Cloudflared has the issue?

4.1.1

### What type of installation are you running?

Home Assistant OS

### Add-on YAML Configuration

```yaml
external_hostname: homeassistant.***
additional_hosts:
  - hostname: plex.***
    service: https://10.1.1.114:32400
  - hostname: bitwarden.***
    service: https://10.1.1.114:7277
tunnel_name: personal-homelab
log_level: debug
tunnel_token: *********
```

### Anything in the logs that might be useful for us?

```txt
HomeAssistant Logs:
2023-04-13 13:35:27.500 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 332, in data_received
    messages, upgraded, tail = self._request_parser.feed_data(data)
  File "aiohttp/_http_parser.pyx", line 551, in aiohttp._http_parser.HttpParser.feed_data
aiohttp.http_exceptions.BadStatusLine: 400, message="Bad status line 'Invalid method encountered'"

Addon Logs:
2023-04-13T11:38:03Z ERR  error="Unable to reach the origin service. The service may be down or it may not be responding to traffic from cloudflared: tls: first record does not look like a TLS handshake" cfRay=7b736ba8d88191e4-FRA ingressRule=0 originService=https://10.1.1.114:8123
2023-04-13T11:38:03Z ERR Request failed error="Unable to reach the origin service. The service may be down or it may not be responding to traffic from cloudflared: tls: first record does not look like a TLS handshake" connIndex=0 dest=https://homeassistant.*****/favicon.ico ip=198.41.192.67 type=http
```

### Steps to reproduce the issue

Just follow the documentation

### Additional information

services stored in additional_hosts works fine

**Comment by elcajon** [comment, marked as answer]
There are a few things that stand out about your configuration that can lead to a problem.

- You use both the configuration options of the add-on for a local tunnel, and the `tunnel_token` which is used for a remote managed tunnel and causes all other configuration options to be ignored. So you should first consider which type of connection you want to use. Take a look [here](https://github.com/brenner-tobias/addon-cloudflared/wiki/How-tos#local-vs-remote-managed-tunnels).
- You have configured the additional_hosts in the add-on and also the homeassistant service in the CF dashboard with https. The error messages seem to indicate that at least HA is not reachable via https, maybe that could already fix your problem.
- You have added 172.30.33.0/24 to the proxied hosts as linked in the instructions, however in the remote managed tunnel UI you are calling a local IP from your home network, this could in context (depending on network configuration) cause the proxy IP not to work. You would probably need to add 127.0.0.1/32. Alternatively you can simply add homeassistant:8123 instead of the IP in the service call.

In general, I would recommend reading the information in the add-on for the documentation of the local managed tunnel as well as the information on the remote managed tunnel and deciding on one of the two.

**Comment by elcajon** [comment]
Since it is not a bug, I am moving the issue to the discussions.

**Comment by maciejbuchert** [comment]
OK. So I've done all your steps and works now. I moved to remote configuration, added additional proxy address and switched https to http in home assistant public hostname. Thanks for help
